{{ cookiecutter.resource_name }}/api.py

```diff
@@ -15,20 +15,54 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-
 """API for manipulating {{ cookiecutter.resource_name }}."""
 
-from __future__ import absolute_import, print_function
+from functools import partial
+
+from .models import {{ cookiecutter.class_name }}Identifier
+from ..fetchers import id_fetcher
+from ..minters import id_minter
+from ..providers import Provider
+from ..api import IlsRecord, IlsRecordsSearch
+
+# provider
+{{ cookiecutter.class_name }}Provider = type(
+    '{{ cookiecutter.class_name }}Provider',
+    (Provider,),
+    dict(identifier={{ cookiecutter.class_name }}Identifier, pid_type='pers')
+)
+# minter
+{{ cookiecutter.name }}_id_minter = partial(id_minter, provider={{ cookiecutter.class_name }}Provider)
+# fetcher
+{{ cookiecutter.name }}_id_fetcher = partial(id_fetcher, provider={{ cookiecutter.class_name }}Provider)
+
 
-from .fetchers import {{ cookiecutter.name }}_id_fetcher
-from .minters import {{ cookiecutter.name }}_id_minter
-from .providers import {{ cookiecutter.class_name }}Provider
-from ..api import IlsRecord
+class {{ cookiecutter.class_name }}sSearch(IlsRecordsSearch):
+    """{{ cookiecutter.class_name }}sSearch."""
 
+    class Meta:
+        """Search only on {{ cookiecutter.name }} index."""
+
+        index = '{{ cookiecutter.resource_name }}'
 
 class {{ cookiecutter.class_name }}(IlsRecord):
     """{{ cookiecutter.class_name }} class."""
 
     minter = {{ cookiecutter.name }}_id_minter
     fetcher = {{ cookiecutter.name }}_id_fetcher
     provider = {{ cookiecutter.class_name }}Provider
+
+    @classmethod
+    def create(cls, data, id_=None, delete_pid=False,
```

Comment:
We can add in the `README` file a note that if `create` and `update` functions are similar to `rero_ils.modules.api` then , these two functions can be safely removed

Reply:
Done